packages/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -501,18 +497,15 @@ export const useGridKeyboardNavigation = (
 
       const viewportPageSize = apiRef.current.getViewportPageSize();
 
-      const getColumnIndexFn = listView ? () => 0 : apiRef.current.getColumnIndex;
+      const getColumnIndexFn = props.listView ? () => 0 : apiRef.current.getColumnIndex;
```

Comment:
Wouldn't it be possible to get rid of the list-view logic here as well? Wouldn't `api.getColumnIndex(params.field)` return the correct index in list-view mode?